### ISSUE TITLE
Bug: Trouble signing up for VZA shifts

### ISSUE THREAD

**Issue by patrickm02L** [cityofaustin/atd-data-tech#10764]
On 11/11/22 Lewis, emailed us that Officer Arevalo, a consistent partner of the OT program, was not seeing any upcoming shifts when he's logged into the app.

Investigate why he's not seeing shifts. Look into whether or not others are not seeing shifts render.

**Comment by patrickm02L** [comment]
Officer A., submitted the following images.

![Image.jpeg](https://images.zenhubusercontent.com/5d68810b05d7792664be403f/df9d9162-6353-41a0-a14d-6ea9b1995767)

![Image (1).jpeg](https://images.zenhubusercontent.com/5d68810b05d7792664be403f/5269f135-31e8-423a-9642-7f862c573a59)

**Comment by patrickm02L** [comment]
We've been updated the shifts calendar, but I was looking at the [Sign Up](https://atd.knack.com/vza#sign-up/?view_476_view=agendaWeek&view_476_date=%222022-11-11T14%3A45%3A53.086Z%22) tab and the shifts are not showing up in the calendar.

**Sign Up** 
<img width="1116" alt="Screen Shot 2022-11-11 at 8 48 50 AM" src="https://user-images.githubusercontent.com/5566727/201365101-bdf60c95-87cd-4a7f-8b6c-ffc4c410184b.png">

**Calendar**

<img width="1157" alt="Screen Shot 2022-11-11 at 8 48 56 AM" src="https://user-images.githubusercontent.com/5566727/201365181-89a68798-e441-4569-b255-4a867e67a200.png">

**Comment by ChrispinP** [comment]
I've checked rules, roles, tasks, and records, everything still appears to be functional within Knack. I'm getting a failure to render from both the URL and the custom JavaScript where the Officer Assignments are not loading for me at all.
![image](https://user-images.githubusercontent.com/67915167/201402582-1607907e-a228-4f19-a5ee-7e7ccbe0660a.png)

I did make a modification to the My Assignments Calendar to show any of the logged in users officer assignments. From glancing at the code, I know its affecting the page and the table view but not sure if its affecting the calendar view as well, quite possibly.
![image](https://user-images.githubusercontent.com/67915167/201403077-59d7a9c6-647a-49c3-a52e-c13adc2ceb4b.png)

**Comment by patrickm02L** [comment]
> I'm getting a failure to render from both the URL and the custom JavaScript where the Officer Assignments are not loading for me at all.

Where are you seeing this? In the Dev Tools or is there a place in the repo that you can point me to?

**Comment by mddilley** [comment]
@patrickm02L @ChrispinP I've been taking a look at this in the test app - `Test - Vision Zero in Action | 3 Feb 2021`. There are some older records in that app, but I made the same change that Chrispin made above to the **Sign Up** calendar view (since it was previously filtering out assignments before the current date), and I am able to see the assignments that I sign up for in the calendar.

**Assignment that I signed up for**
<img width="1200" alt="Screenshot 2022-11-14 at 10 35 06 AM" src="https://user-images.githubusercontent.com/37249039/201720587-7be14e3d-3798-4bfd-9563-53af3baeb380.png">

**My Assignments Calendar**
<img width="1221" alt="Screenshot 2022-11-14 at 10 57 21 AM" src="https://user-images.githubusercontent.com/37249039/201720690-8fb173b1-17f0-4ad8-aac6-ae0f2dc3c333.png">

In the production app, I can't find the new assignments either. It looks like there is a problem with the date filter on the query in the custom code for the assignment records. Working on that now. 

It should request assignments that are available from today to four weeks from today, but it is showing all records (not including the latest additions).

**Comment by mddilley** [comment]
I think that I pinned down the issue and have come up with a patch for it.

Previously: date filtering for what assignments show in the table happened in a request in the custom JS code
The bug: date filtering in that request stopped behaving the way it used to
The result: **all** officer assignments are returned instead of officer assignments that are **today or in the next 28 days**
The fix: update the API view in Knack to do the date filtering instead of the custom JS code

I think that this is a better solution since Knack is less likely to break things that it uses itself. Whatever change happened to the endpoint that we were using doesn't give me a lot of confidence to hack something together that could break again.

To test:
1. Go to https://atd.knack.com/test-vision-zero-in-action-3-feb-2021#sign-up
2. You should see one officer assignment that I manually updated the assignment date to be today, Nov. 14

**Comment by ChrispinP** [comment]
Yea I agree with Mike, the functionality for this page can be built without any custom Code. Filtering & search did just change recently, so those changes from Knack may be part of the blame. Not sure when the page was built but if it was built prior to Knack 2.0 in the spring then a fresh page with correct page rules may do it some good as well.

**Comment by patrickm02L** [comment]
@mddilley thanks for showing this in Dev Sync today! Will approve the changes when assigned to review the PR.

**Comment by mddilley** [comment]
Just applied the fix to the prod API view and code, and I verified that the **Sign up** view shows assignments that are today or in the next 28 days. Also wanted to note that I checked the other date-based filters, and they are working because they are simpler filters.

I also wanted to add @chiaberry's share https://github.com/cityofaustin/atd-knack/pull/124#issuecomment-1314517361 in case we ever have to come back to this issue or if Knack ships a fix.
